api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1673,18 +1673,66 @@ type AWSServiceEndpoint struct {
 }
 
 type AzurePlatformSpec struct {
+	// Credentials is the object containing the Azure credentials needed when creating and managing Azure cloud
+	// infrastructure resources
+	//
+	// +kubebuilder:validation:Required
 	Credentials corev1.LocalObjectReference `json:"credentials"`
-	// The cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
+
+	// Cloud is the cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
+	//
 	// +kubebuilder:validation:Enum=AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud
 	// +kubebuilder:default="AzurePublicCloud"
-	Cloud             string `json:"cloud,omitempty"`
-	Location          string `json:"location"`
+	Cloud string `json:"cloud,omitempty"`
+
+	// Location is Azure region in which the Azure cloud infrastructure resources will be created
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Location is immutable"
```

Comment:
should we mark the fields as immutable instead of this validation
```suggestion
	// +imutable
```

Reply:
I added this 👍🏻